src/pack_unpack.c

```diff
@@ -935,3 +935,21 @@ int json_unpack(json_t *root, const char *fmt, ...) {
 
     return ret;
 }
+
+json_t *json_strtok(const char *str, const char *delim) {
+    json_t *value = NULL;
+    char *s;
+    char *token;
+
+    if (str != NULL) {
+        value = json_array();
+
+        s = strdup(str);
+        for (token = strtok(s, delim); token; token = strtok(NULL, delim)) {
```

Comment:
This should use `strtok_r()` to be safe in a multi-threaded program.

However, I think `json_strtok()` is probably out of scope for this library. (But I'm not a maintainer :)

Reply:
> This should use `strtok_r()` to be safe in a multi-threaded program.

Sure.

> However, I think `json_strtok()` is probably out of scope for this library. (But I'm not a maintainer :)

I dont't know. I needed it, maybe others do too ?

Reply:
Since the purpose of your function is to parse text which is not JSON I would agree that this is out of scope.  That is not a judgement on the potential usefulness, just that jansson is a specialized library rather than a generic library so I think it should not parse non-JSON text.

In addition based on the test you submitted I should point out that `strtok` is not a proper CSV parser which would require processing quoted strings.

Reply:
> In addition based on the test you submitted I should point out that `strtok` is not a proper CSV parser which would require processing quoted strings.

Yeah using CSV was just for the test, it does what `strotk_r()` does.

Feel free to close the MR when you all agree it is out of scope. Thanks for jansson anyway !